### PR TITLE
feat: build pipeline to create new marketplace vhd skus for windows

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

Creating a new build pipeline to create new marketplace skus for aks vhds.
This also creates a build artifact with the sku information needed to publish images to the sku in a release pipeline.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by marosset** [comment]
Part 2 of #2528

File: vhd/publishing/new-windows-sku.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    echo "Could not find sku template file: ${SKU_TEMPLATE_FILE}!"
+    exit 1
+fi
+
+short_date=$(date +"%y%m")
+pretty_date=$(date +"%b %Y")
+
+sku_id="${SKU_PREFIX}-${short_date}"
+
+cat $SKU_TEMPLATE_FILE | sed s/{{ID}}/"$sku_id"/ | sed s/{{MONTH-YEAR}}/"$pretty_date/" > sku.json
```
Comment (marosset) [inline vhd/publishing/new-windows-sku.sh]: Need to fix linting errors here

File: vhd/publishing/windows-sku-template.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "planId": "{{ID}}",
+  "microsoft-azure-corevm.skuTitle": "Windows K8s base image, {{MONTH-YEAR}}",
+  "microsoft-azure-corevm.skuSummary": "Windows K8s base image, {{MONTH-YEAR}}",
+  "microsoft-azure-corevm.hideSKUForSolutionTemplate": true,
+  "microsoft-azure-corevm.hardened": false,
+  "microsoft-azure-corevm.deploymentModels": [
+    "ARM"
+  ],
+  "microsoft-azure-corevm.cloudAvailability": [
```
Comment (marosset) [inline vhd/publishing/windows-sku-template.json]: I will add government and china clouds here

**Comment by marosset** [comment]
I want to do some testing after the windows sku is finishing up a publish operation

File: hack/tools/Makefile [hunk 3]
```diff
@@ -19,6 +19,8 @@ $(LOCALBIN)/ginkgo:
 $(LOCALBIN)/golangci-lint:
 	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(LOCALBIN) v1.21.0
 
+$(LOCALBIN)/pub:
+	GOBIN=$(LOCALBIN) $(GO) get github.com/devigned/pub/...
```
Comment (devigned) [inline hack/tools/Makefile]: Might want to lock this down to a specific version like
```
GOBIN=$(LOCALBIN) $(GO) get github.com/devigned/pub/...@v0.2.0
```

File: .pipelines/publish-sku.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      -w /go/src/github.com/Azure/aks-engine \
+      -e AZURE_TENANT_ID=${AZURE_TENANT_ID} \
+      -e AZURE_CLIENT_ID=${AZURE_CLIENT_ID} \
+      -e AZURE_CLIENT_SECRET=${AZURE_CLIENT_SECRET} \
+      -e SKU_PREFIX=${SKU_PREFIX} \
+      -e SKU_TEMPLATE_FILE=${SKU_TEMPLATE_FILE} \
+      -e PUBLISHER=${PUBLISHER} \
+      -e OFFER=${OFFER} \
+      ${DEIS_GO_DEV_IMAGE} ./vhd/publishing/new-windows-sku.sh
+    displayName: Publishing new SKU
```
Comment (CecileRobertMichon) [inline .pipelines/publish-sku.yaml]: should we avoid the word "publish" since we're not actually hitting the publish button?

Comment (marosset) [reply .pipelines/publish-sku.yaml]: Sure, do you have any suggestions? i couldn't think of a good term to use :(

File: hack/tools/Makefile (hunk 3 above)
Comment (marosset) [reply hack/tools/Makefile]: Sure - i wasn't sure if i needed to add that here as well as in go.mod.

File: .pipelines/publish-sku.yaml (hunk 4 above)
Comment (marosset) [reply .pipelines/publish-sku.yaml]: Create?

Comment (CecileRobertMichon) [reply .pipelines/publish-sku.yaml]: yeah either create or add. Slight preference for create.

File: hack/tools/Makefile (hunk 3 above)
Comment (marosset) [reply hack/tools/Makefile]: updated

File: .pipelines/publish-sku.yaml (hunk 4 above)
Comment (marosset) [reply .pipelines/publish-sku.yaml]: updated to create and also specified marketplace sku in a few spots

**Pull request merged**